File: strategy/ducks/Quack/Squeak.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Text;

namespace ducks.Quack
{
	public class Squeak : IQuackBehaviuor
	{
		public Squeak(StringBuilder stringBuilder) => _stringBuilder = stringBuilder;

		public void Quack() => _stringBuilder?.Append("squeak");

		private readonly StringBuilder _stringBuilder;
	}
}

File: strategy_java/src/ducks/strategies/Squeak.java
package ducks.strategies;

import ducks.base.IQuackBehaviour;

public class Squeak implements IQuackBehaviour {
    public Squeak(StringBuilder sb) {
        _sb = sb;
    }

    @Override
    public void Quack() {
        _sb.append("squeak");
    }

    private final StringBuilder _sb;
}
